pkg/provider/azure_loadbalancer.go

```
@@ -2800,6 +2800,36 @@ func (az *Cloud) getExpectedLBRules(
 	isIPv6 bool,
 ) ([]*armnetwork.Probe, []*armnetwork.LoadBalancingRule, error) {
 	var expectedRules []*armnetwork.LoadBalancingRule
+	// If we are using Pod IP in the LB backend, we skip health probes, disable floating IP and use port.TargetPort.
+	if az.IsLBBackendPoolTypePodIP() {
+		// Check for multi-IP families in the service spec (not allowed for BackendPool of type PodIP).
+		if len(service.Spec.IPFamilies) > 1 {
+			return nil, nil, fmt.Errorf("dual-stack services are not supported when LB backend pool type is PodIP")
+		}
+		// Build rules for each service port but with floatingIP = false, no health probes.
+		for _, port := range service.Spec.Ports {
+			ruleName := az.getLoadBalancerRuleName(service, port.Protocol, port.Port, isIPv6)
+			transportProto, _, _, err := getProtocolsFromKubernetesProtocol(port.Protocol)
+			if err != nil {
+				return nil, nil, fmt.Errorf("failed to parse transport protocol: %w", err)
+			}
+			props, err := az.getExpectedLoadBalancingRulePropertiesForPort(service, lbFrontendIPConfigID, lbBackendPoolID, port, transportProto)
+			if err != nil {
+				return nil, nil, fmt.Errorf("error generating LB rule: %w", err)
+			}
+			// Turn off floating IP and skip health probe attachments.
+			props.EnableFloatingIP = ptr.To(false)
+			props.Probe = nil
+			props.BackendPort = ptr.To(int32(port.TargetPort.IntValue()))
```

I think we won't be able to support named Port (https://kubernetes.io/docs/concepts/services-networking/service/#field-spec-ports). Better to throw a validation error.